### PR TITLE
Add shellcheck to CI

### PR DESCRIPTION
**Pull request by angerman**
This should help ensure we have our shell scripts checked and changes won't introduce unexpected behaviour.

### CODE REVIEW

File: scripts/add-from-github.sh [hunk 1]
```diff
@@ -64,12 +64,12 @@ if [[ ${#SUBDIRS[@]} -gt 1 && (-n $VERSION || -n $REVISION) ]]; then
   exit 1
 fi
 
-if [[ ! "$REPO_URL" =~ "https://github.com/" ]]; then
+if [[ ! "$REPO_URL" =~ https://github.com/ ]]; then
```
Comment (angerman) [inline scripts/add-from-github.sh:67]: Are we sure about this @erikd? @andreabedini do you know?

This was one of the few I was unsure about when reading spellcheck logs, due to the involvement of regex and `/`.

Comment (erikd) [reply scripts/add-from-github.sh:67]: That was the one I was also unsure of, but `shellcheck` recommended it

Comment (andreabedini) [reply scripts/add-from-github.sh:67]:
```
~
❯ { REPO_URL=https://github.com/input-output-hk/cardano-haskell-packages; [[ ! "$REPO_URL" =~ https://github.com/ ]]; echo $? }
1

~
❯ { REPO_URL=https://github.com/input-output-hk/cardano-haskell-packages; [[ ! "$REPO_URL" =~ "https://github.com/" ]]; echo $? }
1
```
it seems to workm the same?

File: scripts/version-lookup.sh [hunk 2]
```diff
@@ -3,18 +3,18 @@
...
 
 versions() {
-  cat dist-newstyle/cache/plan.json \
-    | jq -r '
+  jq -r '
       ."install-plan"[]
       | select(."pkg-src".repo.uri == "https://input-output-hk.github.io/cardano-haskell-packages")
       | {"pkg-name": ."pkg-name", "pkg-version": ."pkg-version"}
-      | @base64' \
+      | @base64' < dist-newstyle/cache/plan.json \
```
Comment (angerman) [inline scripts/version-lookup.sh:17]:
```suggestion
      | @base64' \
      dist-newstyle/cache/plan.json \
```
wouldn't that work just fine? One less redirection to read?

File: scripts/version-lookup.sh [hunk 3]
```diff
@@ -23,5 +23,5 @@ for x in $(versions); do
   pkg_json="$(echo "\"$x\""  | jq -r '@base64d')"
   pkg_name="$(echo "$pkg_json" | jq -r '."pkg-name"')"
   pkg_version="$(echo "$pkg_json" | jq -r '."pkg-version"')"
-  cat "$BASE_DIR/_sources/$pkg_name/$pkg_version/meta.toml" | yj -tj | jq "{\"root\": (. + {\"package\": {\"name\": \"$pkg_name\", \"version\": \"$pkg_version\"}})}"
+  yj -tj < "$BASE_DIR/_sources/$pkg_name/$pkg_version/meta.toml" | jq "{\"root\": (. + {\"package\": {\"name\": \"$pkg_name\", \"version\": \"$pkg_version\"}})}"
```
Comment (angerman) [inline scripts/version-lookup.sh:26]: I think this also works with `yj -tj FILE`?

File: scripts/list-latest-package-versions.sh [hunk 4]
```diff
@@ -4,4 +4,4 @@
 
 REPO=$1
 
-cat "$REPO/foliage/packages.json" | jq -r 'group_by(."pkg-name") | .[] | max_by(."pkg-version") | [."pkg-name", ."pkg-version"] | @tsv' 
+jq -r 'group_by(."pkg-name") | .[] | max_by(."pkg-version") | [."pkg-name", ."pkg-version"] | @tsv' < "$REPO/foliage/packages.json"
```
Comment (angerman) [inline scripts/list-latest-package-versions.sh:7]:
```suggestion
jq -r 'group_by(."pkg-name") | .[] | max_by(."pkg-version") | [."pkg-name", ."pkg-version"] | @tsv' "$REPO/foliage/packages.json"
```

File: scripts/check-archive-extension.sh [hunk 5]
```diff
@@ -57,6 +59,7 @@ NEW_LISTING=$("$TAR" -tvf "$NEW_INDEX_TAR" | sort -k4,5)
 
 DIFF=$(diff <(echo "$OLD_LISTING") <(echo "$NEW_LISTING"))
 
+# shellcheck disable=SC2181
```
Comment (angerman) [inline scripts/check-archive-extension.sh:62]: I think this warrants a comment when disabling shellcheck. Why do we disable it here?

File: scripts/add-from-github.sh (hunk 1 above)
Comment (angerman) [reply scripts/add-from-github.sh:67]: Good!

File: scripts/version-lookup.sh (hunk 2 above)
Comment (erikd) [reply scripts/version-lookup.sh:17]: No, that will not work. `jq` is weird and by default works by reading input from `stdin`.

File: scripts/list-latest-package-versions.sh (hunk 4 above)
Comment (erikd) [reply scripts/list-latest-package-versions.sh:7]: Nope that is not correct. See above.

File: scripts/version-lookup.sh (hunk 3 above)
Comment (erikd) [reply scripts/version-lookup.sh:26]: I do not have `yj` installed locally so I cannot test this, but I do know `yj` is very similar to `jq` in many ways, and `jq` would not work correctly like that.

File: scripts/check-archive-extension.sh (hunk 5 above)
Comment (erikd) [reply scripts/check-archive-extension.sh:66]: Added a full comment there.

**Comment by angerman** [comment]
```
jq FILTER FILE
```
is equivalent to `cat FILE | jq FILTER`, 
however
```
yj FLAGS FILE
```
is _not_ accepted.

Let's stick with `< FILE` for consistency.

**Pull request merged**